package.json

```diff
@@ -2,8 +2,17 @@
   "name": "three",
   "version": "0.136.0",
   "description": "JavaScript 3D library",
-  "main": "build/three.js",
-  "module": "build/three.module.js",
+  "type": "module",
+  "main": "./build/three.js",
+  "module": "./build/three.module.js",
+  "exports": {
+    ".": {
+      "import": "./build/three.module.js",
+      "require": "./build/three.js"
+    },
+    "./examples/jsm/*": "./examples/jsm/*.js",
```

Comment:
What would happen if we renamed this? Say...

```json
"./addons/*": "./examples/jsm/*.js",
```

Would people be able to this?

```js
import { OrbitControls } from 'three/addons/controls/OrbitControls.js'
```

And, would the old version stil work?

```js
import { OrbitControls } from 'three/examples/jsm/controls/OrbitControls.js'
```

Reply:
Apparently yes, you can do that, from the docs:
<img src="https://user-images.githubusercontent.com/7217420/150830858-9e310395-4681-4513-8213-f6f71b8af7ac.png" width="450" />

> And, would the old version stil work?

Unfortunately, no, node (or any bundler) throws an error if you're trying to import something that is not in the `export` fields.

But you can get around this limitation by having both of the import aliases there:

```json
"./addons/*": "./examples/jsm/*.js",
"./examples/jsm/*": "./examples/jsm/*.js",
```

Reply:
Excellent! That makes thing easy.

What do you think about the name `addons`? Do you think @drcmda will like it? 🤓

Reply:
Hahaha I don't mind the name!

I even made a [`three-addons`](https://github.com/marcofugaro/three-addons) package back when `examples/jsm` wasn't a thing 😆

Reply:
Okay, I'll keep it in mind 🙏

Reply:
@mrdoob i do like it 😁